runtime/engine/src/main/java/io/aklivity/zilla/runtime/engine/internal/registry/FileWatcherTask.java

```diff
@@ -0,0 +1,169 @@
+import static java.nio.file.StandardWatchEventKinds.ENTRY_MODIFY;
+import static java.util.concurrent.ForkJoinPool.commonPool;
+import static org.agrona.LangUtil.rethrowUnchecked;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.net.URLConnection;
+import java.nio.file.FileSystems;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.WatchEvent;
+import java.nio.file.WatchKey;
+import java.nio.file.WatchService;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.IntFunction;
+import java.util.function.ToIntFunction;
+
+import org.agrona.ErrorHandler;
+
+import io.aklivity.zilla.runtime.engine.EngineConfiguration;
+import io.aklivity.zilla.runtime.engine.config.KindConfig;
+import io.aklivity.zilla.runtime.engine.config.NamespaceConfig;
+import io.aklivity.zilla.runtime.engine.ext.EngineExtContext;
+import io.aklivity.zilla.runtime.engine.ext.EngineExtSpi;
+import io.aklivity.zilla.runtime.engine.guard.Guard;
+import io.aklivity.zilla.runtime.engine.internal.Tuning;
+
+public class FileWatcherTask extends WatcherTask
+{
+    private final URL configURL;
+    private final Collection<URL> schemaTypes;
+    private final Function<String, Guard> guardsByType;
+    private final ToIntFunction<String> supplyId;
+    private final IntFunction<ToIntFunction<KindConfig>> maxWorkers;
+    private final Tuning tuning;
+    private final Collection<DispatchAgent> dispatchers;
+    private final ErrorHandler errorHandler;
+    private final Consumer<String> logger;
+    private final EngineExtContext context;
+    private final EngineConfiguration config;
+    private final List<EngineExtSpi> extensions;
+    private NamespaceConfig rootNamespace;
+    private byte[] configHash;
+
+    public FileWatcherTask(
+        URL configURL,
+        Collection<URL> schemaTypes,
+        Function<String, Guard> guardsByType,
+        ToIntFunction<String> supplyId,
+        IntFunction<ToIntFunction<KindConfig>> maxWorkers,
+        Tuning tuning,
+        Collection<DispatchAgent> dispatchers,
+        ErrorHandler errorHandler,
+        Consumer<String> logger,
+        EngineExtContext context,
+        EngineConfiguration config,
+        List<EngineExtSpi> extensions)
+    {
+        this.configURL = configURL;
+        this.schemaTypes = schemaTypes;
+        this.guardsByType = guardsByType;
+        this.supplyId = supplyId;
+        this.maxWorkers = maxWorkers;
+        this.tuning = tuning;
+        this.dispatchers = dispatchers;
+        this.errorHandler = errorHandler;
+        this.logger = logger;
+        this.context = context;
+        this.config = config;
+        this.extensions = extensions;
+    }
```

Comment:
This constructor simplifies quite a bit after we move the config changed reactions to outside this class, as mentioned above.